### DISCUSSION TITLE
Mapping 2D Floor Plan in Sionna

### DISCUSSION THREAD

**Discussion by alifyaza** [NVlabs/sionna#500] [General]
I am a student with a keen interest in learning about and working with Sionna by NVIDIA.

I currently have a **2D floor plan** of a building's 2F, which includes specific locations of _Access Points_ that I need to monitor. I would like to know if it is possible to use this mapping in Sionna for monitoring purposes.

Could you please guide me on how to integrate the 2D floor plan with Sionna? Any advice or resources you could provide would be greatly appreciated.

Thank you.

**Comment by jeertmans** [comment]
Hello!
I am not from the Sionna team, but I would recommend you to check if you can convert your 2D floor plan to some 3D map with Blender (free tool).

Then, Sionna's team did a nice tutorial about how to export Blender scenes to Mitsuba compatible file formats (and can be opened with Sionna): https://www.youtube.com/watch?v=7xHLDxUaQ7c. You can skip the part that explains how to import the scene from OpenStreetMap.

Regarding the access points location, the easiest might be to create a custom parser that will read their location from the file and create a transmitter antenna instance for each of them (if that suits your needs).

**Reply by merlinND** [reply]
I agree, this sounds like a good way to do this.
If you prefer, you can also create the Mitsuba scene directly, either as XML or a Python dict. You would have to create the geometry for your scene as well in that case.

**Reply by alifyaza** [reply]
Thank you for your answer @merlinND @jeertmans. I decided to convert the 2D floor plan to 3D model with Blender by following your instructions and the video you provided.

**Reply by merlinND** [reply]
Glad it worked for you, thanks for reporting back!

**Reply by alifyaza** [reply]
Hello @merlinND @jeertmans. I have a question, can I use GANverse3D & NVIDIA Omniverse to create a 3D model from a 2D floor plan and then link it to Sionna NVIDIA? Thank you

**Reply by jeertmans** [reply]
I have never used those, so I cannot help you with that sorry 😅

**Reply by merlinND** [reply]
Hello @alifyaza,

I don't have experience with those either, but I would keep in mind two aspects:
1. Wireless simulation requires (at least for now) fairly simple meshes. AI-based mesh generation tools are likely to generate complex meshes that may not work well for simulation.
2. Omniverse doesn't have an exporter to the Mitsuba scene format used by Sionna, so you would have to convert from one format to another either way.
